src/widget/wtrackmenu.cpp

```diff
@@ -2808,9 +2809,34 @@ void WTrackMenu::addToAutoDJ(PlaylistDAO::AutoDJSendLoc loc) {
         return;
     }
 
+    // If the "Replace" action was clicked and there are tracks in the
+    // AutoDJ queue, ask for confirmation.
     PlaylistDAO& playlistDao = m_pLibrary->trackCollectionManager()
                                        ->internalCollection()
                                        ->getPlaylistDAO();
+    const auto tracks = playlistDao.getAutoDJTrackIds();
+    if (loc == PlaylistDAO::AutoDJSendLoc::REPLACE &&
+            tracks.size() > 0) {
```

Comment:
```suggestion
    if (s_confirmForAutoDjReplace && loc == PlaylistDAO::AutoDJSendLoc::REPLACE &&
            tracks.size() > 0) {
```
otherwise the "Don't ask again" checkbox won't do anything...